lib/Doctrine/ORM/Mapping/ArrayAccessImplementation.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+trait ArrayAccessImplementation
+{
+    /** @param array-key $offset */
```

Comment:
All of the consumers of this trait use string offsets, so we might as well…

```suggestion
    /** @param string $offset */
```

Apart from that, PHPStan and Psalm should support generic traits.